pkg/agent/metrics/prometheus.go

```diff
@@ -0,0 +1,119 @@
+// Copyright 2020 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metrics
+
+import (
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"github.com/vmware-tanzu/antrea/pkg/agent/interfacestore"
+	"k8s.io/klog"
+	"net/http"
+	"os"
+	"strconv"
+
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/vmware-tanzu/antrea/pkg/agent/openflow"
+)
+
+type OVSStatManager struct {
+	ofClient     openflow.Client
+	OVSBridge    string
+	OVSTableDesc *prometheus.Desc
+}
+
+func (c *OVSStatManager) OVSGetStatistics() (
```

Comment:
>>@srikartati I thought that it's better to separate Prometheus and the monitor code (for CRD) as they might evolve in different directions, and we could end up with one cluttered package.
We already have a followup PR (#325) and other could/should follow.

Agree that having separate metrics components for Prometheus integration is better for future metrics. I just wondered about using the querier interface rather than defining them again.
However, looking at it more, for every poll/scrape cycle, we seem to run the functions GetContainerInterfaceNum() and GetOVSFlowTable() again and again. Rather, should we use the Inc, Dec, Set etc. of metrics at appropriate places to decrease the complexity. For example, incrementing pod num metric in connectInterfaceToOVS and decrementing in disconnectInterfaceFromOVS. Similarly for OVS flow count. This could be useful when there are more and more Antrea agent metrics. This can be a follow up change/enhancement. Let me know your comments.

Reply:
As far as I understand these values are cached so there's no sense in caching these again in Prometheus client.
If there is any impact though, I'd do so in a separate change.